data-prepper-plugins/rds-source/src/main/java/org/opensearch/dataprepper/plugins/source/rds/datatype/impl/NumericTypeHandler.java

```diff
@@ -0,0 +1,55 @@
+package org.opensearch.dataprepper.plugins.source.rds.datatype.impl;
+
+import org.opensearch.dataprepper.plugins.source.rds.datatype.DataTypeHandler;
+import org.opensearch.dataprepper.plugins.source.rds.model.TableMetadata;
+import org.opensearch.dataprepper.plugins.source.rds.datatype.MySQLDataType;
+
+import java.math.BigInteger;
+
+public class NumericTypeHandler implements DataTypeHandler {
+
+    @Override
+    public String handle(final MySQLDataType columnType, final String columnName, final Object value,
+                         final TableMetadata metadata) {
+        if (value == null) {
+            return null;
+        }
+
+        if (!columnType.isNumeric()) {
+            throw new IllegalArgumentException("ColumnType is not numeric: " + columnType);
+        }
+
+        if (!(value instanceof Number)) {
+            throw new IllegalArgumentException("Value is not a number: " + value);
+        }
+
+        return handleNumericType(columnType, (Number) value);
+    }
+
+    private String handleNumericType(final MySQLDataType columnType, final Number value) {
```

Comment:
Are we converting all numeric types to String type here? For numbers that OpenSearch supports, should we keep them as numbers?